lib/matplotlib/backends/backend_tkagg.py

```diff
@@ -731,11 +730,10 @@ def set_message(self, s):
 
     def draw_rubberband(self, event, x0, y0, x1, y1):
         height = self.canvas.figure.bbox.height
-        y0 =  height-y0
-        y1 =  height-y1
-        try: self.lastrect
-        except AttributeError: pass
-        else: self.canvas._tkcanvas.delete(self.lastrect)
+        y0 = height - y0
+        y1 = height - y1
+        if hasattr(self, "lastrect"):
```

Comment:
This is a hot-path for UI,  Under normal usage I think we expect this to exist for often that in does not so the `try..except` pattern is marginally faster.

```python

def tester(obj):
    if hasattr(obj, 'string'):
        pass

def tryer(obj):
    try:
        obj.string
    except:
        pass


class A:
    def __init__(self):
        self.string = 'foo'

class B:
    ...
```

```python
In [61]: %timeit tester(1)
459 ns ± 3.11 ns per loop (mean ± std. dev. of 7 runs, 1000000 loops each)

In [62]: %timeit tryer(1)
524 ns ± 0.955 ns per loop (mean ± std. dev. of 7 runs, 1000000 loops each)

In [63]: 
In [64]: a = A()

In [65]: %timeit tryer(a)
123 ns ± 0.202 ns per loop (mean ± std. dev. of 7 runs, 10000000 loops each)

In [66]: %timeit tester(a)
194 ns ± 0.281 ns per loop (mean ± std. dev. of 7 runs, 10000000 loops each)

In [67]: 
In [68]: b = B()

In [69]: %timeit tryer(b)
488 ns ± 1.11 ns per loop (mean ± std. dev. of 7 runs, 1000000 loops each)

In [70]: %timeit tester(b)
423 ns ± 2.14 ns per loop (mean ± std. dev. of 7 runs, 1000000 loops each)

In [74]: 
```